demos/cms/cms_ddec.c

```diff
@@ -34,7 +34,8 @@ int main(int argc, char **argv)
 
     rcert = PEM_read_bio_X509(tbio, NULL, 0, NULL);
 
-    BIO_reset(tbio);
+    if (BIO_reset(tbio))
+        goto err;
```

Comment:
This should be `if (BIO_reset(tbio) < 0)`

Reply:
okay.  I will update those checks.

I find `BIO_reset()` strange b/c its return value depends on what type of BIO is being reset:

https://www.openssl.org/docs/manmaster/man3/BIO_reset.html

> BIO_reset() normally returns 1 for success and <=0 for failure. File BIOs are an exception, they return 0 for success and -1 for failure.

Reply:
We use the boolean checks only for functions that return 0 or 1 only. As BIO reset is different (and awkward as you noticed) comparison operator must be used appropriately.

Reply:
fixed.